run.py

```diff
@@ -197,6 +197,104 @@ def get_files(args, dirname, min_size=1, min_date=''):
+
+            if ".pair1." in potential_input:
+                tmp_fq_outputs.append(tmp_fq_output.replace(".pair1.", ".pair2."))
+                inputs.append(potential_input.replace(".pair1.", ".pair2."))
+            elif ".pair2." in potential_input:
+                # Ribodetector handles pair1 and pair2 together.
+                continue
+
+            with tempdir("ribocounts", sample + " inputs") as workdir:
+                for input_fname in inputs:
+                    subprocess.check_call([
+                        "aws", "s3", "cp", "%s/%s/cleaned/%s" % (
+                            S3_BUCKET, args.bioproject, input_fname), input_fname])
+
+                    # Ribodetector gets angry if the .fq extension isn't in the filename 
+                    os.rename(input_fname, input_fname.replace(".gz", ".fq.gz"))
+
+                # Add .fq extensions to input files
+                inputs = [i.replace(".gz", ".fq.gz") for i in inputs]
+
+                # Compute average read lengths. For paired-end reads, average length is 
+                # computed only from pair1 reads.
+                print("Calculating average read length...")
+                def calculate_average_read_length(file_path):
+                    total_len = 0
+                    total_reads = 0
+                    with gzip.open(file_path, 'rt') as inf:
+                        for (title, sequence, quality) in FastqGeneralIterator(inf):
+                            total_len += len(sequence)
+                            total_reads += 1
+                    return total_reads, round(total_len / total_reads)
+
+                total_reads, avg_length = calculate_average_read_length(inputs[0])
+                sample_read_count += total_reads
+                print("Done. Average read length is ", avg_length)
+
+                ribodetector_cmd = [
+                    "ribodetector_cpu",
+                    "--ensure", "rrna",
+                    "--threads", "24"
+                    ]
+                ribodetector_cmd.extend(["--len", str(avg_length)])
+
+                ribodetector_cmd.append("--input")
+                ribodetector_cmd.extend(inputs)
+                ribodetector_cmd.append("--output")
+                ribodetector_cmd.extend(tmp_fq_outputs)
+
+                subprocess.check_call(ribodetector_cmd)
+                
+                # Count the number of reads passing rRNA screening (i.e. classified as 
+                # non-rRNA)
+                def count_reads_in_fastq(file_path):
+                    total_reads = 0
+                    with open(file_path, 'r') as inf:
+                        for (title, sequence, quality) in FastqGeneralIterator(inf):
+                            total_reads += 1
+                    return total_reads
+                
+                # Count non-rRNA reads in sample. Paired-end reads are counted once
+                sample_nonrrna_count += count_reads_in_fastq(tmp_fq_outputs[0])
```

Comment:
You could do:

```
total_nonrrna_reads, _ = calculate_average_read_length(tmp_fq_outputs[0])
sample_nonrrna_count += total_nonrrna_reads
```

Then you don't need `count_reads_in_fastq`, though it's slightly less clear.

Reply:
I agree the current formulation is a bit clunky but I'm going to leave as is. One reason: `calculate_average_read_length` opens a gzip file where the output of RiboDetector is just a fastq file.